### ISSUE TITLE
Testing KSP processors

### ISSUE THREAD

**Issue by sgammon** [google/ksp#1207]
Hey there KSP authors,

Is there an easy way to generate the inputs for a processor, and test its behavior? I.e. can the writing of files be mocked and inspected easily?

Thank you in advance for any help you can offer on this subject

**Comment by neetopia** [comment]
Unfortunately from KSP itself there isn't such support, there are some KSP processors with official support which you might find some insight on how they test their processors for production.

**Comment by ting-yuan** [comment]
Let's keep this issue open as a reminder to refine KSP docs.

**Comment by sgammon** [comment]
@ting-yuan agreed, and it may end up that Google should provide its own testing functionality... `kotlin-compile-testing` has fallen woefully behind and [does not yet support `1.8`](kotlin-compile-testing#344).

**Comment by GuilhE** [comment]
> [@ting-yuan](https://github.com/ting-yuan) it may end up that Google should provide its own testing functionality... `kotlin-compile-testing` has fallen woefully behind and [does not yet support `1.8`](https://github.com/tschuchortdev/kotlin-compile-testing/pull/344).

Adding that `tschuchortdev/kotlin-compile-testing` (supports 1.9/2.0/2.1) and `ZacSweers/kotlin-compile-testing` (supports KSP2) do not support KMP it would be very beneficial if Google or Jetbrains could help us out with their own testing library.

**Comment by WildOrangutan** [comment]
I'm pretty disappointed on lack of testing guidance for such a core Kotlin library.
 
- `tschuchortdev/kotlin-compile-testing` is not maintained anymore
- `ZacSweers/kotlin-compile-testing` fork is also not actively maintained anymore ([the author is 'winding down'](https://github.com/ZacSweers/kotlin-compile-testing/issues/340))

**Comment by jcopenhop** [comment]
Agreed. we are moving to the zacsweers fork so we can at least get on kotlin2 but that won't work forever.
